jdaviz/configs/default/tests/test_data_menu.py

```diff
@@ -91,3 +110,84 @@ def test_data_menu_create_subset(imviz_helper):
     dm.create_subset('circle')
     assert imviz_helper.app.session.edit_subset_mode.edit_subset == []
     assert imviz_helper.viewers['imviz-0']._obj.toolbar.active_tool_id == 'bqplot:truecircle'
+
+
+def test_data_menu_remove_subset(specviz_helper, spectrum1d):
+    # load 2 data entries
+    specviz_helper.load_data(spectrum1d, data_label="test")
+    new_spec = specviz_helper.get_spectra(apply_slider_redshift=True)["test"]*0.9
+    specviz_helper.load_data(new_spec, data_label="test2")
+
+    dm = specviz_helper.viewers['spectrum-viewer']._obj.data_menu
+    sp = specviz_helper.plugins['Subset Tools']
+
+    sp._obj.import_region(SpectralRegion(6000 * spectrum1d.spectral_axis.unit,
+                                         6100 * spectrum1d.spectral_axis.unit),
+                          combination_mode='new')
+    sp._obj.import_region(SpectralRegion(6000 * spectrum1d.spectral_axis.unit,
+                                         6100 * spectrum1d.spectral_axis.unit),
+                          combination_mode='new')
+
+    assert dm.layer.choices == ['test', 'test2', 'Subset 1', 'Subset 2']
+    dm.layer.selected = ['Subset 1']
+    dm.remove_from_viewer()
+
+    # subset visibility is set to false, but still appears in menu (unlike removing data)
+    assert dm.layer.choices == ['test', 'test2', 'Subset 1', 'Subset 2']
+    assert dm._obj.layer_items[2]['label'] == 'Subset 1'
+    # TODO: sometimes appearing as mixed right now, known bug
+    assert dm._obj.layer_items[2]['visible'] is not True
+
+    # selection should not have changed by removing subset from viewer
+    assert dm.layer.selected == ['Subset 1']
+    dm.remove_from_app()
+    # TODO: not quite sure why this isn't passing, seems to
+    # work on local tests, so may just be async?
+    # assert dm.layer.choices == ['test', 'test2', 'Subset 2']
```

Comment:
I'm not sure if the cause of this is the same as the other known state delayed bugs or not, but do see the choice dissappear when testing locally in the notebook, so I suspect this is probably caused by the async nature of `app.remove_data_from_viewer`.  We can either track that down here or consider it as a follow-up since I don't _think_ its directly related to any new code added in this PR.

Reply:
In the notebook, I don't see the selected subset disappear from the menu after removing from app until after I click somewhere else - that's probably why this is failing.

Reply:
Actually, the same thing happens with remove from viewer - the eye icon doesn't change to toggled invisible until after I click somewhere else.

Reply:
ok, probably related to the subset state bug you're investigating then, do you think?  We can probably re-enable this as a regression test for that then?

Reply:
Hmm, likely. I think that's an ok plan for now.